contracts/interstake-yield-generator/src/contract.rs

```diff
@@ -538,6 +657,71 @@ mod execute {
             .add_attribute("release_timestamp", release_timestamp.to_string())
             .add_messages(undelegate_msgs))
     }
+
+    /// updates the allowed address list with the given address and expiration
+    pub fn update_allowed_address(
+        deps: DepsMut,
+        env: Env,
+        info: MessageInfo,
+        address: String,
+        expiration: Option<u64>,
+    ) -> Result<Response, ContractError> {
+        let config = CONFIG.load(deps.as_ref().storage)?;
+        if config.owner != info.sender {
+            return Err(ContractError::Unauthorized {});
+        }
```

Comment:
Since this message is going to be called by DAO, maybe it should allow to whitelist addresses in a bulk.

Reply:
I think its alright like this. This is only called when a new partneship is enabled. i dont think that will happen that oftne